### PR TITLE
Lazyloading the Map

### PR DESCRIPTION
**Pull request by TheKeithStewart**

### CODE REVIEW

File: src/app/esri-map/esri-map.component.ts [hunk 1]
```diff
@@ -1,20 +1,19 @@
...
-let _map;
-
 @Component({
   selector: 'app-esri-map',
   templateUrl: './esri-map.component.html',
   styleUrls: ['./esri-map.component.css'],
   providers: [EsriLoaderService]
 })
 export class EsriMapComponent implements OnInit {
+  map: any;
```
Comment (tomwayson) [inline src/app/esri-map/esri-map.component.ts:11]: Ah ha! I bet this is what fixed the issue w/ the map route only working the first time.

**Review by tomwayson — COMMENTED** [review]
@kgs916 

This is awesome. I pulled and it works perfectly.

Only issue is that neither the unit nor the e2e tests pass (mostly b/c the dom has changed and they haven't been updated). If you're interested in fixing them you can just push another commit to this branch and I'll merge.

Otherwise, I'm glad to merge as is and I can fix them after.

Just let me know and thanks again for this great contribution!

File: src/app/esri-map/esri-map.component.ts (hunk 1 above)
Comment (TheKeithStewart) [reply src/app/esri-map/esri-map.component.ts:11]: That is correct.  This allows the state of the map to be cached.

**Comment by TheKeithStewart** [comment]
I'm working on fixing up the tests.  I've done a bit of Angular 2 development but I'm much more familiar with testing Angular 1.  I should have this PR ready soon though.

**Comment by tomwayson** [comment]
Great. Thanks. 

FYI - I had to make these changes https://github.com/tomwayson/angular-cli-esri/commit/af36a3f43c151522cc7414a723f4ae7cda87257b to get the tests working after I generated the component and service.

**Comment by TheKeithStewart** [comment]
Thanks. I've got the unit tests working now. Had to jump through a bunch of
hoops for Angular Material. Once I get the e2e tests working I will resend
the PR.

**Comment by TheKeithStewart** [comment]
The unit and e2e tests are not working again.

**Comment by tomwayson** [comment]
Well done @kgs916 - the material integration would have thrown me for a loop for sure.

Thanks again for taking this on. Great job!

**Pull request merged**